File: 4.Counting Element - MissingInteger/answer.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(A):
    # write your code in Python 3.6
    numbers = {i: None for i in range(1, len(A)+1)}
    for i in A:
        if(i in numbers):
            del numbers[i]
    
    if(len(numbers) == 0):
        return len(A)+1
    return min(numbers)

File: 4.Counting Element - MissingInteger/answer.js
// you can write to stdout for debugging purposes, e.g.
// console.log('this is a debug message');

function solution(A) {
    // write your code in JavaScript (Node.js 8.9.4)
    let ans = {}
    for (val of A){
        ans[val] = undefined
    }
    let index = 1
    for (key in ans) {
        if(key > index)
            return index
        else if(index > key)   
            continue
        else 
            index++
    }
    return index
}
